### PR TITLE
[Task][Integration] Integrate display  tenant list

### PR DESCRIPTION
**Pull request by alOletres**
# Issue Link

[Link to the corresponding issue in the github taskboard](https://github.com/orgs/Web-Masters-Freelancing/projects/1/views/1?pane=issue&itemId=53620321)

# Definition of Done
- [x] Users can display a tenant list thru server data

# Notes
None

# Pre-conditions
None

# Expected Output
Able to display tenant list

# Screenshots/Recordings
image

### CODE REVIEW

File: server/src/_gen-prisma-classes/room.ts [hunk 1]
```diff
@@ -14,6 +15,8 @@ export class Room {
   description?: string;
 
   @ApiProperty({ type: Number })
+  @IsNotEmpty()
```
Comment (LodiAniez) [inline server/src/_gen-prisma-classes/room.ts:18]: ayaw iedit ang genfile bayot, auto generated mana. Pag create ug custom dto file unya ioverride lang na nga property para mabutangan nimo ug decorator ([How to override parent class properties](https://stackoverflow.com/questions/70434444/override-parent-class-properties-typescript))

Comment (LodiAniez) [reply server/src/_gen-prisma-classes/room.ts:18]: ang reason nganong dili dapat iedit ang gen file kay everytime morun ang prisma generator, tanan nimong giedit mawala to, kinahanglan napod nimo imanual ug edit tanan files

File: server/src/user/user.controller.ts [hunk 2]
```diff
@@ -12,4 +14,18 @@ export class UserController {
...
+    status: 200,
+    type: AllUsersResponseDto,
+    isArray: true,
+  })
+  @Get('/all')
+  async getUsers(): Promise<AllUsersResponseDto[]> {
+    const users = await this.userService.fetchUsers();
+    console.log('users', users);
+
+    return users;
```
Comment (LodiAniez) [inline server/src/user/user.controller.ts:29]:
```suggestion
    return await this.userService.fetchUsers();
```

File: server/src/user/user.service.ts [hunk 3]
```diff
@@ -67,4 +68,35 @@ export class UserService {
 
     return user;
   }
+
+  async fetchUsers(): Promise<AllUsersResponseDto[]> {
```
Comment (LodiAniez) [inline server/src/user/user.service.ts:72]:
```suggestion
  async findAll(): Promise<AllUsersResponseDto[]> {
```

File: server/src/user/user.controller.ts [hunk 4]
```diff
@@ -12,4 +14,18 @@ export class UserController {
...
     await this.userService.create(payload);
   }
+
+  @ApiExtraModels(AllUsersResponseDto)
+  @ApiResponse({
+    status: 200,
+    type: AllUsersResponseDto,
+    isArray: true,
+  })
+  @Get('/all')
```
Comment (LodiAniez) [inline server/src/user/user.controller.ts:24]:
```suggestion
  @Get('/')
```

File: server/src/user/dto/fetch-users.dto.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { PickType } from '@nestjs/swagger';
+import { User as UserEntity } from '../../_gen-prisma-classes/user';
+
+export class AllUsersResponseDto extends PickType(UserEntity, [
```
Comment (LodiAniez) [inline server/src/user/dto/fetch-users.dto.ts:4]:
```suggestion
export class FindAllUsersResponseDto extends PickType(UserEntity, [
```

File: web/src/app/tenants/hooks.ts [hunk 6]
```diff
@@ -197,44 +206,46 @@ export const useHook = () => {
...
+        if (value.room) {
+          const rooms = value.room;
+          value = { ...value, ...rooms };
+        }
+
+        return value;
+      });
 
-  const handleEdit = (values: TenantFormValues | undefined) => {
+      console.log("usersData", usersData);
```
Comment (LodiAniez) [inline web/src/app/tenants/hooks.ts:231]: iremove tanan console log bur

File: web/src/app/tenants/hooks.ts [hunk 7]
```diff
@@ -197,44 +206,46 @@ export const useHook = () => {
...
+
       setTitle("EDIT TENANT");
       setBtnName("Save Changes");
-      setInitialValues({ ...values });
+      setInitialValues({
+        id,
+        name,
+        contact,
+        address,
+        roomId: roomId ?? 0,
```
Comment (LodiAniez) [inline web/src/app/tenants/hooks.ts:247]: itest ni ug sure bur, kay ang `roomId` required ni sa `Billable` na model, ayaw inania, if undefined ang `roomId`, ayaw pacreata ug tenant, pag return ug error

File: web/src/hooks/api/user.ts [hunk 8]
```diff
@@ -4,6 +4,12 @@ import { CreateUserDto } from "@/store/api/gen/user";
 export const useUserApi = () => {
   const [create] = userApi.useUserControllerCreateMutation();
 
+  const {
+    isFetching: isFetchingUsers,
+    isError: isUsersError,
```
Comment (LodiAniez) [inline web/src/hooks/api/user.ts:9]:
```suggestion
    isError: isErrorFetchingUsers,
```

File: web/src/app/tenants/hooks.ts [hunk 9]
```diff
@@ -15,6 +15,20 @@ interface RoomHistoryValues {
   userId: number;
 }
 
+interface BillableValues {
+  roomId: number;
+  room: Pick<RoomsFormValues, "type">;
+}
+
+interface TenantValues
```
Comment (LodiAniez) [inline web/src/app/tenants/hooks.ts:23]: ireuse ang type na gi export sa genfile bur

**Review by LodiAniez — CHANGES_REQUESTED** [review]

File: server/src/_gen-prisma-classes/room.ts (hunk 1 above)
Comment (alOletres) [reply server/src/_gen-prisma-classes/room.ts:18]: nalimot diay kog kuha sorry

Comment (alOletres) [reply server/src/_gen-prisma-classes/room.ts:18]: dli man mawala baby girl

Comment (alOletres) [reply server/src/_gen-prisma-classes/room.ts:18]: ako na ge restart og pa run naa raman

**Pull request closed without merging**